### PR TITLE
diagnostics: list diagnostic names in long desc

### PR DESCRIPTION
**Pull request by sosiouxme**
Fixes https://github.com/openshift/origin/issues/5984

It looks like

```
Diagnostics may be individually run with --diagnostics. The available
diagnostic names are:
AnalyzeLogs,ClusterRegistry,ClusterRoleBindings,ClusterRoles,ClusterRouter,ConfigContexts,MasterConfigCheck,MasterNode,NodeConfigCheck,NodeDefinitions,UnitStatus
```

Would I like the list to be more aesthetically pleasing? Sure. Do I really wanna implement that? No. If there's a nice text-formatting module somewhere, great...

### CODE REVIEW

**Comment by sosiouxme** [comment]
@liggitt PTAL and merge...

File: pkg/cmd/experimental/diagnostics/diagnostics.go [hunk 1]
```diff
@@ -85,7 +90,7 @@ func NewCommandDiagnostics(name string, fullName string, out io.Writer) *cobra.C
 	cmd := &cobra.Command{
 		Use:   name,
 		Short: "This utility helps you troubleshoot and diagnose.",
-		Long:  fmt.Sprintf(longDescription, fullName),
+		Long:  fmt.Sprintf(longDescription, fullName, strings.Join(availableDiagnostics().List(), ",")),
```
Comment (liggitt) [inline pkg/cmd/experimental/diagnostics/diagnostics.go]: join with ", "

File: pkg/cmd/experimental/diagnostics/diagnostics.go [hunk 2]
```diff
@@ -71,6 +72,10 @@ you will receive an error if they are not found. For example:
 * If a client config file is not found, client and cluster diagnostics
   are skipped.
 
+Diagnostics may be individually run with --diagnostics. The available
+diagnostic names are:
+%[2]v
```
Comment (liggitt) [inline pkg/cmd/experimental/diagnostics/diagnostics.go]: `%[2]s`, since you're making it a string

**Comment by liggitt** [comment]
nits, LGTM otherwise

File: pkg/cmd/experimental/diagnostics/diagnostics.go (hunk 1 above)
Comment (sosiouxme) [inline pkg/cmd/experimental/diagnostics/diagnostics.go]: It looks better but then you can't cut and paste with `-d`

**Comment by sosiouxme** [comment]
[merge]

**Comment by sosiouxme** [comment]
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin/8070/ flaked with https://github.com/openshift/origin/issues/6444

**Comment by sosiouxme** [comment]
Let's try another [merge]

**Comment by sosiouxme** [comment]
flipping [merge] already

**Comment by sosiouxme** [comment]
[merge] i say

**Comment by sosiouxme** [comment]
thpbpbpt [merge]

**Comment by sosiouxme** [comment]
now is the time for all good bots to [merge]

**Comment by sosiouxme** [comment]
test works how come you won't [merge]

**Comment by sosiouxme** [comment]
[merge] and accept no failure!

**Comment by deads2k** [comment]
re[merge]

**Pull request merged**